v2/dtos/requests/event.go

```diff
@@ -23,14 +23,11 @@ type AddEventRequest struct {
 	Event              dtos.Event `json:"event" validate:"required"`
 }
 
-// NewAddRequest creates, initializes and returns an AddEventRequest which has no readings
-// Sample usage:
-//    request := NewAddRequest("myProfile", "myDevice")
-//    request.Event.AddSimpleReading("myInt32Resource", v2.ValueTypeInt32, int32(1234))
-func NewAddRequest(profileName string, deviceName string) AddEventRequest {
+// NewAddEventRequest creates, initializes and returns an AddEventRequests
+func NewAddEventRequest(event dtos.Event) AddEventRequest {
```

Comment:
please just rename it.  the signature is fine as is.
you can still modify the necessary after using this factory method